src/databricks/labs/ucx/source_code/python_ast_util.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import annotations
+
+import ast
+from dataclasses import dataclass
+
+
+@dataclass
+class Span:
+    """Represents a (possibly multiline) source code span."""
+
+    start_line: int
+    start_col: int
+    end_line: int
+    end_col: int
+
+
+class AstUtil:
+    @staticmethod
+    def extract_callchain(node: ast.AST) -> ast.Call | None:
```

Comment:
isn't it the same as `databricks.labs.ucx.source_code.python_linter.ASTLinter.locate`?

Reply:
It doesn't seem to work the same way - locate wants to be given the full form of the callchain, whereas we don't know that form here. We just know there is somewhere in the callchain a call to "saveToTable", but not what comes before it (could be anything).